sklearn/utils/testing.py

```diff
@@ -52,13 +52,38 @@
 from sklearn.utils._joblib import joblib
 from sklearn.utils._unittest_backport import TestCase
 
+additional_names_in_all = []
+try:
+    from nose.tools import raises as _nose_raises
+    deprecation_message = (
```

Comment:
Use a try-except-else rather than doing this in the try block (and rather than doing a second try block below)

Reply:
This is an existing code, I removed it because it had obsolete nose code then I rolled back the changes on @rth [comment](https://github.com/scikit-learn/scikit-learn/pull/12974#issuecomment-453863621) as the deprecation said code is slated for removal only in version 0.22.